dali/ft/filecopy.cpp

```diff
@@ -2596,6 +2651,345 @@ void FileSprayer::pushParts()
+    }
+};
+
+class AzureFileClient : public CApiCopyClient
+{
+    std::unique_ptr<Shares::ShareFileClient> fileClient;
+    std::unique_ptr<Shares::StartFileCopyOperation> fileCopyOp;
+public:
+    ApiCopyStatus startCopy(const char *target, const char * source)
+    {
+        try
+        {
+            fileClient.reset(new Shares::ShareFileClient(target));
+            fileCopyOp = std::make_unique<Shares::StartFileCopyOperation>(std::move(fileClient->StartCopy(source)));
+            status = ApiCopyStatus::Pending;
+        }
+        catch (const Azure::Core::RequestFailedException& e)
+        {
+            IERRLOG("AzureFileClient startCopy failed: %s (code %d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
+            status = ApiCopyStatus::Failed;
+            throw MakeStringException(DFTERR_CopyFailed, "%s", e.ReasonPhrase.c_str());
+        }
+        return status;
+    }
+    ApiCopyStatus getProgress(CDateTime & dateTime, int64_t & outputLength)
+    {
+        if (status!=ApiCopyStatus::Pending)
+            return status;
+        try
+        {
+            fileCopyOp->Poll();
+            Shares::Models::FileProperties props = fileCopyOp->Value();
+            dateTime.setString(props.LastModified.ToString().c_str());
+            outputLength = props.FileSize;
+            Shares::Models::CopyStatus tstatus = props.CopyStatus.HasValue()?props.CopyStatus.Value():(Shares::Models::CopyStatus::Pending);
+            if (tstatus==Shares::Models::CopyStatus::Success) // FYI. CopyStatus is an object so can't use switch statement
+                status = ApiCopyStatus::Success;
+            else if (tstatus==Shares::Models::CopyStatus::Pending)
+                status = ApiCopyStatus::Pending;
+            else if (tstatus==Shares::Models::CopyStatus::Aborted)
+                status = ApiCopyStatus::Aborted;
+            else
+                status = ApiCopyStatus::Failed;
+        }
+        catch(const Azure::Core::RequestFailedException& e)
+        {
+            IERRLOG("Transfer using API .Poll() failed: %s (code %d)", e.ReasonPhrase.c_str(), static_cast<int>(e.StatusCode));
+            status = ApiCopyStatus::Failed;
+        }
+        return status;
+    }
+    ApiCopyStatus abortCopy()
+    {
+        if (status==ApiCopyStatus::Aborted || status==ApiCopyStatus::Failed)
+            return status;
+        int64_t outputLength;
+        CDateTime dateTime;
+        status = getProgress(dateTime, outputLength);
+        switch (status)
+        {
+            case ApiCopyStatus::Pending:
+                try
+                {
+                    if (fileCopyOp->HasValue() && fileCopyOp->Value().CopyId.HasValue())
+                        fileClient->AbortCopy(fileCopyOp->Value().CopyId.Value().c_str());
+                    else // not sure that this can ever happen
+                        IERRLOG("Transfer using API .AbortCopy() failed: unexpected response from API");
```

Comment:
The error message does not really correspond to the test - more like a missing copyId.